modules/internal/DefaultSparse.chpl

```diff
@@ -235,7 +235,114 @@ module DefaultSparse {
         add_help(ind);
       }
     }
-  
+
+    proc bulkAdd(inds: [] index(rank, idxType), sorted=false, noDuplicate=false){
+      var numAdded = inds.size; //maybe remove this var
+
+      //find individual insert points
+      //and eliminate duplicates between inds and dom
+      var indivInsertPts: [{0..#numAdded}] int;
+      var actualInsertPts: [{0..#numAdded}] int; //where to put in newdom
+
+      if !sorted then QuickSort(inds);
+
+      //eliminate duplicates --assumes sorted
+      if !noDuplicate {
+        //make sure lastInd != inds[inds.domain.low]
+        var lastInd = inds[inds.domain.low] + 1; 
+        for (i, p) in zip(inds, indivInsertPts)  {
+          if i == lastInd then p = -1;
+          else lastInd = i;
+        }
+      }
+
+      //verify sorted and no duplicates if not --fast
+      if boundsChecking {
+        VerifySort(inds, "bulkAdd: Data is not sorted, call the function with \
+            sorted=false");
+
+        //check duplicates assuming sorted
+        const indsStart = inds.domain.low;
+        const indsEnd = inds.domain.high;
+        var lastInd = inds[indsStart];
+        for i in indsStart+1..indsEnd {
+          if inds[i] == lastInd then halt("There are duplicates"); 
```

Comment:
How about ?

bulkAdd: There are duplicates, call the function with unique=false